packages/@lwc/babel-plugin-component/src/post-process/dedupe-imports.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+module.exports = function dedupeImports(path) {
+    const body = path.get('body');
+    const importStatements = body.filter((s) => s.isImportDeclaration());
+    const visited = new Map();
+    importStatements.forEach((importPath) => {
```

Comment:
so, if I understand, this will just group them together, even though they were produced as separated by a previous plugin, is that it?

Reply:
if yes, I wonder how to avoid grouping user's defined imports, vs compiler defined imports?

Reply:
Thats the point @leobalter raised earlier. We do already have linting rules for avoid this. This is just for our own internals. Making the transforms "context aware" its tricky and order might vary, hence this deduping at the end of the process.